extendeddatasquare_test.go

```diff
@@ -426,6 +426,34 @@ func TestEquals(t *testing.T) {
 	})
 }
 
+func TestRootsHelper(t *testing.T) {
```

Comment:
[nit] the `Helper` suffix doesn't seem necessary. Commonly if a function name is `Foo` then the corresponding test will be named `TestFoo`

```suggestion
func TestRoots(t *testing.T) {
```

Reply:
ohh I see why you named this `TestRootsHelper` to avoid a naming collision with the existing `TestRoots`. Okay let's disregard my suggestion for now. In a follow up PR we can rename `TestRoots` => `Test_getRoots` because it's testing the private functions `getRowRoots` and `getColRoots`. Then we can rename this function `TestRoots`.